### ISSUE TITLE
prefer-named-capture-group is reporting incorrect locations

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#12233]
**Tell us about your environment**

* **ESLint Version:**  6.3.0
* **Node Version:** 10.16.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parserOptions: {
    ecmaVersion: 2015,
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

[Demo link](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgcHJlZmVyLW5hbWVkLWNhcHR1cmUtZ3JvdXA6IFwiZXJyb3JcIiovXG5cblJlZ0V4cChcIlxcXFxkXFxcXHMoYSlcIik7XG5cblJlZ0V4cChcIihcIiArIFwiYVwiICsgXCIpXCIpO1xuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo1LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImNvbnN0cnVjdG9yLXN1cGVyIjoyLCJmb3ItZGlyZWN0aW9uIjoyLCJnZXR0ZXItcmV0dXJuIjoyLCJuby1hc3luYy1wcm9taXNlLWV4ZWN1dG9yIjoyLCJuby1jYXNlLWRlY2xhcmF0aW9ucyI6Miwibm8tY2xhc3MtYXNzaWduIjoyLCJuby1jb21wYXJlLW5lZy16ZXJvIjoyLCJuby1jb25kLWFzc2lnbiI6Miwibm8tY29uc3QtYXNzaWduIjoyLCJuby1jb25zdGFudC1jb25kaXRpb24iOjIsIm5vLWRlYnVnZ2VyIjoyLCJuby1kZWxldGUtdmFyIjoyLCJuby1kdXBlLWFyZ3MiOjIsIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6Miwibm8tZHVwZS1rZXlzIjoyLCJuby1kdXBsaWNhdGUtY2FzZSI6Miwibm8tZW1wdHkiOjIsIm5vLWVtcHR5LWNoYXJhY3Rlci1jbGFzcyI6Miwibm8tZW1wdHktcGF0dGVybiI6Miwibm8tZXgtYXNzaWduIjoyLCJuby1leHRyYS1ib29sZWFuLWNhc3QiOjIsIm5vLWV4dHJhLXNlbWkiOjIsIm5vLWZhbGx0aHJvdWdoIjoyLCJuby1mdW5jLWFzc2lnbiI6Miwibm8tZ2xvYmFsLWFzc2lnbiI6Miwibm8taW5uZXItZGVjbGFyYXRpb25zIjoyLCJuby1pbnZhbGlkLXJlZ2V4cCI6Miwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2UiOjIsIm5vLW1peGVkLXNwYWNlcy1hbmQtdGFicyI6Miwibm8tbmV3LXN5bWJvbCI6Miwibm8tb2JqLWNhbGxzIjoyLCJuby1vY3RhbCI6Miwibm8tcHJvdG90eXBlLWJ1aWx0aW5zIjoyLCJuby1yZWRlY2xhcmUiOjIsIm5vLXJlZ2V4LXNwYWNlcyI6Miwibm8tc2VsZi1hc3NpZ24iOjIsIm5vLXNoYWRvdy1yZXN0cmljdGVkLW5hbWVzIjoyLCJuby1zcGFyc2UtYXJyYXlzIjoyLCJuby10aGlzLWJlZm9yZS1zdXBlciI6Miwibm8tdW5kZWYiOjIsIm5vLXVuZXhwZWN0ZWQtbXVsdGlsaW5lIjoyLCJuby11bnJlYWNoYWJsZSI6Miwibm8tdW5zYWZlLWZpbmFsbHkiOjIsIm5vLXVuc2FmZS1uZWdhdGlvbiI6Miwibm8tdW51c2VkLWxhYmVscyI6Miwibm8tdW51c2VkLXZhcnMiOjIsIm5vLXVzZWxlc3MtY2F0Y2giOjIsIm5vLXVzZWxlc3MtZXNjYXBlIjoyLCJuby13aXRoIjoyLCJyZXF1aXJlLWF0b21pYy11cGRhdGVzIjoyLCJyZXF1aXJlLXlpZWxkIjoyLCJ1c2UtaXNuYW4iOjIsInZhbGlkLXR5cGVvZiI6Mn0sImVudiI6e319fQ==)

```js
/*eslint prefer-named-capture-group: "error"*/

RegExp("\\d\\s(a)");

RegExp("(" + "a" + ")");
```

```bash
eslint index.js
```

**What did you expect to happen?**

2 errors, but not with incorrect locations.

**What actually happened? Please include the actual, raw output from ESLint.**

```
3:13  error  Capture group '(a)' should be converted to a named or non-capturing group  prefer-named-capture-group
5:9   error  Capture group '(a)' should be converted to a named or non-capturing group  prefer-named-capture-group
```

```js
/*eslint prefer-named-capture-group: "error"*/

RegExp("\\d\\s(a)"); // highlighting '\s('

RegExp("(" + "a" + ")"); // highlighting '(" '
```

**Are you willing to submit a pull request to fix this bug?**

Yes.

Perhaps if `arguments[0]` is not a string literal which `raw` is same as internal presentation, then report the location of the whole `arguments[0]`?

**Comment by mdjermanovic** [comment]
Marking this as accepted because this small bug is obvious.

I'll fix this, just need help with what would be a better solution:

1. Report the exact location of the disallowed group when it's possible, otherwise the whole `arguments[0]` node. 
2. Or, always report the full location of the `arguments[0]` node, just to be consistent.
3. Or, always report the whole NewExpression/CallExpression node, that's how other regexp rules already work.

**Comment by g-plane** [comment]
I prefer Solution 1.

**Comment by platinumazure** [comment]
As I noted in the PR, I think I might (only slightly, not strongly) prefer solution 2. If we didn't specify the capture group content in the message, I would be strongly in favor of solution 1, but as it is, I think solution 2 would be a good experience for most users.
